core/src/main/java/com/linecorp/armeria/server/http/dynamic/DynamicMethod.java

```diff
@@ -0,0 +1,101 @@
+package com.linecorp.armeria.server.http.dynamic;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Parameter;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.linecorp.armeria.annotations.PathVariable;
+
+public final class DynamicMethod {
+    /**
+     * Returns {@link DynamicMethod} from given method object. If there is any problem, e.g., not all
+     * parameters of the method is not annotated with {@link PathVariable} or deserializable type, raises
+     * {@link IllegalArgumentException}.
+     * */
+    public static DynamicMethod of(Object object, Method method) {
```

Comment:
Could we also add support for asynchronous methods? For example, we could detect a method that returns a `CompletionStage` or a `CompletableFuture`, and run it without leaving the I/O thread if so.